### PR TITLE
use project instead of transform to access contravariant components

### PR DESCRIPTION
**Pull request by simonbyrne**
This should help avoid some unnecessary ops and checks

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by charleskawczynski** [comment]
I can go ahead and rebase this to double check if this improves anything

**Review by charleskawczynski — APPROVED** [review]

File: src/Geometry/conversions.jl [hunk 1]
```diff
@@ -141,21 +141,18 @@ LocalVector(u::ContravariantVector{<:Any, (3,)}, ::LocalGeometry{(1, 2)}) =
     CovariantVector(u, local_geometry).u₃
 
 @inline contravariant1(u::AxisVector, local_geometry::LocalGeometry) =
-    transform(Contravariant123Axis(), u, local_geometry).u¹
+    @inbounds project(Contravariant1Axis(), u, local_geometry)[1]
```
Comment (charleskawczynski) [inline src/Geometry/conversions.jl:144]: Wait, why was this changed from `Contravariant123Axis` to `Contravariant1Axis`? I assume this is equivalent? cc @simonbyrne

Comment (simonbyrne) [reply src/Geometry/conversions.jl:144]: We only care about the 1st component
